homeworks/anatoliy.pogoriliy_AyzekUorren/homework_6/task.js

```diff
@@ -0,0 +1,144 @@
+
+function User() {
+    let localFullName = null;
+    let localPhone = null;
+
+    Object.defineProperty(this, 'fullName', {
+        get() {
+            return localFullName
+                .replace(/\w*/g, word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase());
+        },
+        set(value) {
+            localFullName = value;
+        },
+    });
+
+    Object.defineProperty(this, 'phone', {
+        get() {
+            return localPhone.replace(/\D/g, '');
+        },
+        set(value) {
+            localPhone = value;
+        },
+    });
+}
+
+function randomHorseCredentials() {
+    const arrayHorseBreeds = [
+        'Arabian', 'Quarter Horse', 'Thoroughbred',
+        'Tennessee Walker', 'Morgan', 'Paint',
+        'Appaloosa', 'Miniature Horse', 'Warmblood',
+        'Andalusian',
+    ];
+    const arrayHorseNames = [
+        'Velvet', 'Kingston', 'Sniper',
+        'Sugar Stallion', 'Barq', 'Joker', 'Preacher',
+        'Flash Sparkle', 'Quinn', 'Jill',
+        'Flash Sparks', 'Justice', 'Bruno',
+        'Rogue', 'Talzanna', 'Leonardo',
+        'Orion', 'Narnia', 'Sweet Sparks', 'Kalli',
+    ];
+    const breed = arrayHorseBreeds[Math.floor(Math.random() * (arrayHorseBreeds.length))];
+    const name = arrayHorseNames[Math.floor(Math.random() * (arrayHorseBreeds.length))];
+    return { name, breed };
+}
+
+class Horse {
+    constructor(name, breed) {
+        this.localName = name;
+        this.localBreed = breed;
+    }
+
+    get name() {
+        return this.localName;
+    }
+
+    get breed() {
+        return this.localBreed;
+    }
+}
+
+class Racer extends Horse {
+    constructor(...args) {
+        super(...args);
+        this.setSpeed();
+        this.localDsitance = 0;
+    }
+
+    get speed() {
+        return this.localSpeed;
+    }
+
+    get distance() {
+        return this.localDsitance;
+    }
+
+    setSpeed() {
+        this.localSpeed = Math.floor(Math.random() * (15 - 10 + 1) + 10);
```

Comment:
`15 - 10 + 1` quals `6` ;)

Reply:
This if I or other will forget limit's.)